components/ca-remote-image-capture.js

```diff
@@ -0,0 +1,159 @@
+                bubbles: true
+            });
+            this.dispatchEvent(event);
+        }
+
+        /**
+         * @description Executes when any attribute is changed on the element
+         * @param {string} attrName - the name of the attribute to have changed
+         * @param {string} oldVal - the old value of the attribute
+         * @param {string} newVal - the new value of the attribute
+         * @returns {undefined}
+         */
+        attributeChangedCallback(attrName, oldVal, newVal) {
+            // Only need to re-render the component if the attrbute value has changed
+            if (oldVal !== newVal) {
+                this.render.call(this);
+            }
+        }
+
+        /** @property {Integer} ca-image-capture.height - gets the height */
+        get height() {
+            return parseInt(this.getAttribute('height') || '720', 10);
+        }
+
+        /** @param {Integer} value - sets the height */
+        set height(value) {
+            this.setAttribute('height', value || 720);
+        }
+
+        /** @property {Integer} ca-image-capture.width - gets the width */
+        get width() {
+            return parseInt(this.getAttribute('width') || '1280', 10);
+        }
+
+        /** @param {Integer} value - sets the height */
+        set width(value) {
+            this.setAttribute('width', value || 1280);
+        }
+
+        /** @property {Boolean} ca-image-capture.imageCaptured - gets the captured state */
+        get imageCaptured() {
+            return (this.getAttribute('image-captured') === 'true');
+        }
+
+        /** @param {Boolean} value - sets the captured state */
+        set imageCaptured(value) {
+            this.setAttribute('image-captured', value === true);
+        }
+
+        /** @property {String} ca-image-capture.streamUrl - sets the stream url */
+        get streamUrl() {
+            return this.imageStreamUrl;
+        }
+
+        /** @param {String} value - sets the stream url */
+        set streamUrl(value) {
+            this.imageStreamUrl = value;
+        }
+
+        /** @property {String} ca-image-capture.captureUrl - gets the capture url */
+        get captureUrl() {
+            return this.imageCaptureUrl;
+        }
+
+        /** @param {String} value - sets the capture url */
+        set captureUrl(value) {
+            this.imageCaptureUrl = value;
+        }
+
+        /** @property {String} ca-image-capture.value - gets the image base 64 val */
+        get value() {
+            if (this.imageCaptured) {
+                return this.imageCanvas.toDataURL('image/jpeg', 0.8).replace('data:image/jpeg;base64,', '');
+            }
+            return '';
+        }
+
+        /**
+         * @description Renders the element (invoked via render.call(this))
+         * @returns {undefined}
+         */
+        render() {
+            // only render based on these conditions.
+            if (this.streamUrl || !this.imageCaptured) {
```

Comment:
Hello :)

Sorry, I couldn't help myself. For what it's worth - (!this.streamUrl || this.imageCaptured) has been reversed to get rid of the early return. It **should** operate in much the same way. Since an operation is only carried out (this.streamUrl || !this.imageCaptured) is true.